monai/data/dataset.py

```diff
@@ -180,13 +180,13 @@ def _pre_transform(self, item_transformed):
             random transform object
 
         """
-        if not isinstance(self.transform, Compose):
-            raise ValueError("transform must be an instance of monai.transforms.Compose.")
-        for _transform in self.transform.transforms:
+        for _transform in self.transform.transforms:  # type:ignore
             # execute all the deterministic transforms
             if isinstance(_transform, Randomizable) or not isinstance(_transform, Transform):
                 break
-            item_transformed = apply_transform(_transform, item_transformed)
+            # this is to be consistent with CacheDataset even though it's not in a multi-thread situation.
+            _xform = deepcopy(_transform) if isinstance(_transform, ThreadUnsafe) else _transform
```

Comment:
I think `deepcopy` a class instance maybe a little bit dangerous as some kinds of class properties may not support pickle? For example:
```py
import numpy as np

class Transform:
    def __init__(self):
        self.np = np

a = A()
a_copy = copy.deepcopy(a)
```
Got error: `can’t pickle module objects error`

So maybe we need a test method to guarantee all the exiting and future `thread-unsafe` transforms can support `deepcopy`?

Thanks.

Reply:
for now we only have one such threadunsafe transform... 
https://github.com/Project-MONAI/MONAI/blob/2fb4b2170506da73effd82f4acab5fedb30722e5/monai/transforms/spatial/array.py#L491

let me think about this, and perhaps will create a ticket about it. Thanks!

Reply:
I know we don't have many thread-unsafe transforms so far, but need a test or method to make sure future thread-unsafe transforms can support `deepcopy`.

Thanks.